### PR TITLE
Update example python files to use sphinx-gallery

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
Most of the code is the same as it was. The main thing was just adding the code to a function `main()` and some docstrings to explain what the code is doing.

### CODE REVIEW

File: examples/__init__.py
Comment (EdwardSafford-NOAA) [inline examples/__init__.py:1]: This is intentionally empty, right?

Comment (CoryMartin-NOAA) [reply examples/__init__.py:1]: yeah the `__init__` file is needed to `import` the directory

File: examples/histograms/README.txt
Comment (EdwardSafford-NOAA) [inline examples/histograms/README.txt:1]: Is this a placeholder for future expansion?

File: examples/map_plots/custom_map_domain.py [hunk 1]
```diff
@@ -1,31 +1,51 @@
+"""
+Create a custom map domain
+--------------------------
+
+EMCPy's map domain includes a feature
+to create a custom dictionary with specified
+lat and lon inputs to create a specific map
+domain. This is helpful for users who do want
```
Comment (EdwardSafford-NOAA) [inline examples/map_plots/custom_map_domain.py]: Suggestion on phrasing "This is helpful for users who want to look at a specific domain that is not included in EMCPy."

**Review by EdwardSafford-NOAA — APPROVED** [review]
I look forward to seeing how this works.  Looks super useful.  I had a few questions I noted separately but no requested changes.

General question -- the several `.png` files won't change, right?  I got myself in significantly warm water for adding a binary file (tar in my case) that changed very infrequently, and learned how much fun it is to extract a binary file from (the free version) of github.

Also I assume the several, largely empty, `README.txt` files are included as placeholders for future expansion.

File: examples/histograms/README.txt
Comment (kevindougherty-noaa) [reply examples/histograms/README.txt:1]: image
This is what the mostly empty README.txt files are for. They are just mainly to create a title in the examples gallery page

**Pull request merged**